disttask/importinto/encode_and_sort_operator.go

```diff
@@ -116,6 +126,40 @@ func (op *encodeAndSortOperator) Done() <-chan struct{} {
 type chunkWorker struct {
 	ctx context.Context
 	op  *encodeAndSortOperator
+
+	dataWriter  *external.EngineWriter
+	indexWriter *importer.IndexRouteWriter
+}
+
+func newChunkWorker(ctx context.Context, op *encodeAndSortOperator, workerID int64) *chunkWorker {
+	w := &chunkWorker{
+		ctx: ctx,
+		op:  op,
+	}
+	if op.tableImporter.IsGlobalSort() {
+		// sorted index kv storage path: /{taskID}/{subtaskID}/index/{indexID}/{workerID}
+		indexWriterFn := func(indexID int64) *external.Writer {
+			builder := external.NewWriterBuilder().
+				SetOnCloseFunc(func(summary *external.WriterSummary) {
```

Comment:
I plan to let the "WriterSummary" result be returned when Writer.Close, not passing it around using OnClose. What do you think?

Reply:
lgtm